flopy/mf6/mfmodel.py

```diff
@@ -1937,3 +1943,12 @@ def plot(self, SelPackList=None, **kwargs):
         )
 
         return axes
+
+    @staticmethod
+    def _resolve_idomain(idomain, botm):
+        if idomain is None:
+            if botm is None:
```

Comment:
Not sure I understand why there is a dependence of `idomain` on `botm`. Do we want to retain `idomain=None` if `botm` is not defined. Wouldn't it be better from the `modelgrid` creation to fail if botm has not been defined when creating a discretization instance?

Reply:
Our current approach with `modelgrid` creation is to allow for the creation of a `modelgrid` without top and botm.  This has always been allowed so that a user can make use of much of the `modelgrid` functionality without using top and botm.  Also, a `modelgrid` is requested by flopy whenever grid data is loaded and the "check" option requested, including when top and botm are loaded.  Failing to create `modelgrid` when botm is still in the process of loading would break our current approach.  I am open to reconsidering this approach, but I thought that was beyond the scope of this PR, which is to fix a very specific problem.

Regarding the dependence of initializing the default `idomain` (of all 1's) on botm, to initialize the default `idomain` one needs to first know the shape of the model grid.  The `modelgrid` class currently has only one variable, besides `idomain`, that defines the full shape of the grid, `botm`.  The simplest, and hopefully least prone to error, implementation was therefore to get the shape of `idomain` from the `botm` array.

However, on further considering there is a potential problem with this.  The model grid could get cached with the default `idomain` values when flopy loads `botm` and then not get updated when `idomain` is finally loaded.  Going to add some code to fix this potential bug.